runtime/reconcilers/canvas.go

```diff
@@ -122,3 +129,74 @@ func (r *CanvasReconciler) checkAnyComponentHasValidSpec(ctx context.Context, re
 	}
 	return false, nil
 }
+
+func (r *CanvasReconciler) validateMetricsViewTimeConsistency(ctx context.Context, refs []*runtimev1.ResourceName) error {
+	metricsViews := make(map[string]*runtimev1.Resource)
+	for _, ref := range refs {
+		if ref.Kind != runtime.ResourceKindComponent {
+			continue
+		}
+		component, err := r.C.Get(ctx, ref, false)
+		if err != nil {
+			continue
+		}
+		var mvName string
+		validSpec := component.GetComponent().State.ValidSpec
+		if validSpec != nil && validSpec.RendererProperties != nil {
+			for k, v := range validSpec.RendererProperties.Fields {
+				if k == "metrics_view" {
+					mvName = v.GetStringValue()
+					break
+				}
+			}
+		}
```

Comment:
Leverage the components' `refs` instead of parsing the properties. For example:
```
	for _, ref := range component.Meta.Refs {
		if ref.Kind != runtime.ResourceKindMetricsView {
			continue
		}
		...
	}
```